plugins/processors/ec2tagger/ec2tagger.go

```diff
@@ -327,7 +322,17 @@ func (t *Tagger) Start(ctx context.Context, _ component.Host) error {
 			Region:    t.ec2MetadataRespond.region,
 		}
 		t.ec2API = t.ec2Provider(ec2CredentialConfig)
-		go func() { //Async start of initial retrieval to prevent block of agent start
+
+		if _, ok := t.ec2API.(*ec2.EC2); ok {
```

Comment:
You can just use the result of the type assertion so you don't need to assert it again later.

```suggestion
		if client, ok := t.ec2API.(*ec2.EC2); ok {
```